alibi_detect/utils/frameworks.py

```diff
@@ -15,3 +18,91 @@
     has_sklearn = True
 except ImportError:
     has_sklearn = False
+
+
+HAS_BACKEND = {
+    'tensorflow': has_tensorflow,
+    'sklearn': has_sklearn,
+    'pytorch': has_pytorch,
+}
+
+
+def _iter_to_str(iterable):
+    """ Correctly format iterable of items to comma seperated sentence string."""
+    items = [f'`{option}`' for option in iterable]
+    last_item_str = f'{items[-1]}' if not items[:-1] else f' and {items[-1]}'
+    return ', '.join(items[:-1]) + last_item_str
+
+
+class BackendValidator:
+    def __init__(self, backend_options: Dict[Optional[str], List[str]], construct_name: str):
```

Comment:
btw I'm not sure `mypy` actually typechecks functions/methods with missing return type hints, can you verify and add `None` return types where necessary?

Reply:
My understanding is that if a method is missing all type hints, so for arguments and return types then `mypy` doesn't type check it. So if `__init__` doesn't require any parameters then you have to specify the return type as `None` or it won't be checked. This isn't an issue here but might be elsewhere. I'll make sure to change it where necessary.